cmd/emulator/Dockerfile

```diff
@@ -20,8 +21,15 @@ ARG TARGETARCH
 RUN --mount=type=ssh \
     --mount=type=cache,target=/go/pkg/mod \
     --mount=type=cache,target=/root/.cache/go-build \
-    GO111MODULE=on CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH \
-    go build -ldflags "-extldflags -static" -o ./app ./cmd/emulator
+    if [ "$TARGETARCH" = "arm64" ] ; then \
+    GO111MODULE=on CGO_ENABLED=1 CGO_FLAGS="-O2 -D__BLST_PORTABLE__" \
+    GOOS=$TARGETOS GOARCH=$TARGETARCH CC=aarch64-linux-gnu-gcc \
+    go build -ldflags "-extldflags -static" -o ./app ./cmd/emulator ; \
+    elif [ "$TARGETARCH" = "amd64" ] ; then \
+    GO111MODULE=on CGO_ENABLED=1 CGO_FLAGS="-O2 -D__BLST_PORTABLE__" \
+    GOOS=$TARGETOS GOARCH=$TARGETARCH CC=x86_64-linux-gnu-gcc \
+    go build -ldflags "-extldflags -static" -o ./app ./cmd/emulator ; \
+    fi
```

Comment:
It looks like the only difference between the two commands for the two architectures is the CC flag?
Maybe just have `export CC=...` in the branches, and run the command shared between after the if-statement

Reply:
I've tried to implement this without success:
- Docker does not persist shell's data from one `RUN` to the next
- For persistent data between `RUN` commands, we can use `ARG` or `ENV`, but you can't update them with shell on the Dockerfile itself 🤷🏼‍♂️ 

Let me know if you have ideas. Ideally, these if else statements should live in shell before calling `docker build`, but here our CI.yml relies on a tool that manages the `docker build` call.

Reply:
Ah, OK. I'm not a Docker expert, maybe ops can help with this

Reply:
maybe @sjonpaulbrown or @peterargue have an idea on how to set the `CC` flag within Docker so that it persists in the following commands?

Reply:
what about something like this?
```dockerfile
ARG TARGETOS 
ARG TARGETARCH
ENV GO111MODULE=on \
	CGO_ENABLED=1 \
	CGO_FLAGS="-O2 -D__BLST_PORTABLE__" \
	GOOS=$TARGETOS \
	GOARCH=$TARGETARCH
RUN --mount=type=ssh \
    --mount=type=cache,target=/go/pkg/mod \
    --mount=type=cache,target=/root/.cache/go-build \
    if [ "$TARGETARCH" = "arm64" ] ; then \
        export CC=aarch64-linux-gnu-gcc; \
    elif [ "$TARGETARCH" = "amd64" ] ; then \
        export CC=x86_64-linux-gnu-gcc; \
    fi; \
    go build -ldflags "-extldflags -static" -o ./app ./cmd/emulator ; \
```

Reply:
It works well! The `ENV` variables becomes env variables within the docker image but that's fine I guess. 
Thanks @peterargue!

Reply:
@turbolent the PR should be ready now, thanks!